compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -368,6 +405,150 @@ void LinalgFusePass::runOnOperation() {
     return signalPassFailure();
   }
 }
+namespace {
+/// Pattern to wrap splitReduction transformation.
+struct CodegenSplitReduction
+    : public OpInterfaceRewritePattern<linalg::LinalgOp> {
+  CodegenSplitReduction(MLIRContext *context, bool reductionReordering,
+                        int64_t size, linalg::LinalgTransformationFilter filter,
+                        PatternBenefit benefit = 1)
+      : OpInterfaceRewritePattern<linalg::LinalgOp>(context, benefit),
+        reductionReordering(reductionReordering),
+        size(size),
+        filter(std::move(filter)) {}
+
+  LogicalResult matchAndRewrite(linalg::LinalgOp op,
+                                PatternRewriter &rewriter) const override {
+    // Make sure that
+    // - the pass has not been applied before
+    // - has tensor semantics
+    // - number of reduction loops == 1
+    // - has exactly 1 output
+    // - index map has only projected permutations
+    // - is a linalg generic op
+    // - has exactly 1 input
+    // - if enableReductionReordering is not set, then operand is an int
+    // - innermost dimension of the input operand is reduction
```

Comment:
[optional] the failure messages already tells us the check. The comments are redundant to me. I do know some people like to write such comments, so I'm not marking the change required. Just an FYI that I've seen style guide discussions in Google that asking people don't write redundant comments. The best is to make them self-describing.

Reply:
I needed the comment to write the code to not lose track of all the checks.